src/libstd/sys/common/wtf8.rs

```diff
@@ -206,19 +205,10 @@ impl Wtf8Buf {
     /// Copied from String::push
     /// This does **not** include the WTF-8 concatenation check.
     fn push_code_point_unchecked(&mut self, code_point: CodePoint) {
-        let cur_len = self.len();
-        // This may use up to 4 bytes.
-        self.reserve(4);
-
-        unsafe {
-            // Attempt to not use an intermediate buffer by just pushing bytes
-            // directly onto this string.
-            let slice = slice::from_raw_parts_mut(
-                self.bytes.as_mut_ptr().offset(cur_len as isize), 4
-            );
-            let used = encode_utf8_raw(code_point.value, slice).unwrap();
-            self.bytes.set_len(cur_len + used);
-        }
+        let bytes = unsafe {
+            char::from_u32_unchecked(code_point.value).encode_utf8()
```

Comment:
Ok. But I’m asking, because I really don’t know: do we need to? Is this one of those things that appear to work when you try it but is technically undefined behavior and can cause bad things in future compiler versions?

I still have nigthly CI for https://github.com/SimonSapin/rust-wtf8 which has similar code, failing now that `encode_utf8_raw` is gone, and I’m unsure what to do about it. And I imagine other libraries may be tempted to do something similar: I recall the `regex` crate doing some contortions internally to avoid creating a surrogate `char` when "iterating" over character ranges like `[a-z]`.